xrviz/dashboard.py

```diff
@@ -46,72 +51,88 @@ def __init__(self, data):
         self.kwargs = self.control.kwargs
         self.var = self.kwargs['Variables']
-        if self.is_dataset:
-            self.var_dims = list(self.data[self.var].dims)
-        else:
-            self.var_dims = list(self.data.dims)
-
-        #  var_selector_dims refers to dims for which index_selectors would be created
-        not_to_index = [self.kwargs['x'], self.kwargs['y'], *self.kwargs['dims_to_agg']]
-        self.var_selector_dims = sorted([dim for dim in self.var_dims if dim not in not_to_index])
-
         self.index_selectors = []
         self.output[1].clear()  # clears Index_selectors
 
-        for dim in self.var_selector_dims:
-            if self.is_dataset:
-                ops = list(self.data[self.var][dim].values)
-            else:
-                ops = list(self.data[dim].values)
-
-            if self.kwargs[dim] == 'Select':
-                selector = pn.widgets.Select(name=dim, options=ops)
-            else:
-                selector = pn.widgets.DiscretePlayer(name=dim, value=ops[0], options=ops)
-
-            self.index_selectors.append(selector)
-            selector.param.watch(self.callback_for_indexed_graph, ['value'], onlychanged=False)
-
-        self.output[0] = self.create_indexed_graph()
-        for selector in self.index_selectors:
-            if isinstance(selector, pn.widgets.Select):
-                self.output[1].append(selector)
-            else:
-                player = player_with_name_and_value(selector)
-                self.output[1].append(player)
-
-    def check_is_plottable(self, var):
-        """
-        If a variable is 1-d, disable plot_button for it.
-        """
-        self.plot_button.disabled = False  # important to enable button once disabled
         if self.is_dataset:
-            var = var[0]
-            if len(list(self.data[var].dims)) <= 1:
-                self.plot_button.disabled = True
-        else:
-            if len(self.data.dims) <= 1:
-                self.plot_button.disabled = True
+            x = self.kwargs['x']
+            if not self.is_non_indexed_coord(x):  # i.e is a var_dim
+                self.var_dims = list(self.data[self.var].dims)
+                #  var_selector_dims refers to dims for which index_selectors would be created
+                self.var_selector_dims = self.kwargs['dims_to_select_animate']
+                self.index_selectors = []
+                self.output[1].clear()  # clears Index_selectors
+
+                for dim in self.var_selector_dims:
+                    ops = list(self.data[self.var][dim].values)
+
+                    if self.kwargs[dim] == 'Select':
+                        selector = pn.widgets.Select(name=dim, options=ops)
+                    else:
+                        selector = pn.widgets.DiscretePlayer(name=dim, value=ops[0], options=ops)
+                    self.index_selectors.append(selector)
+                    self._register(selector, selector.name)
```

Comment:
This clears out previous watchers and callbacks for a widget of the same name? It might be nice to clear all events before starting to add new ones.

Reply:
Since we can't access the old widgets, once plot button is clicked, it would be better to simply delete those. I have implemented this deletion of old widgets.

Reply:
When you say "delete", you mean removing from the `_sigs` and `_map` dicts? Perhaps Siglot should grow a method for doing this.